coderd/rbac/scopes_catalog.go

```diff
@@ -0,0 +1,87 @@
+package rbac
+
+import (
+	"sort"
+	"strings"
+)
+
+// externalLowLevel is the curated set of low-level scope names exposed to users.
+// Any valid resource:action pair not in this set is considered internal-only
+// and must not be user-requestable.
+var externalLowLevel = map[ScopeName]struct{}{
+	// Workspaces
+	"workspace:read":                {},
+	"workspace:create":              {},
+	"workspace:update":              {},
+	"workspace:delete":              {},
+	"workspace:ssh":                 {},
+	"workspace:start":               {},
+	"workspace:stop":                {},
+	"workspace:application_connect": {},
+	"workspace:*":                   {},
+
+	// Templates
+	"template:read":   {},
+	"template:create": {},
+	"template:update": {},
+	"template:delete": {},
+	"template:use":    {},
+	"template:*":      {},
+
+	// API keys (self-management)
+	"api_key:read":   {},
+	"api_key:create": {},
+	"api_key:update": {},
+	"api_key:delete": {},
+	"api_key:*":      {},
+
+	// Files
+	"file:read":   {},
+	"file:create": {},
+	"file:*":      {},
+
+	// Users (personal profile only)
+	"user:read_personal":   {},
+	"user:update_personal": {},
+
+	// User secrets
+	"user_secret:read":   {},
+	"user_secret:create": {},
+	"user_secret:update": {},
+	"user_secret:delete": {},
+	"user_secret:*":      {},
+}
```

Comment:
I wonder if we will need these as constants later.

We can adjust if required though

Reply:
Yeah, I thought about it too, but the keys in [RBACPermissions](https://github.com/coder/coder/blob/HEAD/coderd/rbac/policy/policy.go#L67-L67) aren't typed either. We'd probably need to define those as constants first. Still, I agree that a const enum + action enum would be better here.